src/Compilers/CSharp/Test/Symbol/Symbols/RequiredMembersTests.cs

```diff
@@ -404,65 +408,106 @@ [RequiredMember] C
         );
     }
 
-    [Fact]
-    public void RequiredMemberAttributeEmitted_OverrideRequiredProperty_MissingRequiredOnOverride01()
+    [Theory]
+    [CombinatorialData]
+    public void RequiredMemberAttributeEmitted_OverrideRequiredProperty_MissingRequiredOnOverride01(bool useMetadataReference)
     {
-        var comp = CreateCompilationWithRequiredMembers(@"
-class Base
+        var @base = @"
+public class Base
 {
     public virtual required int Prop { get; set; }
-}
+}";
+
+        var derived = @"
 class Derived : Base
 {
     public override int Prop { get; set; }
 }
-");
+";
+
+        var comp = CreateCompilationWithRequiredMembers(@base + derived);
 
         comp.VerifyDiagnostics(
             // (8,25): error CS9501: 'Derived.Prop': cannot remove 'required' from 'Base.Prop' when overriding
             //     public override int Prop { get; set; }
             Diagnostic(ErrorCode.ERR_OverrideMustHaveRequired, "Prop").WithArguments("Derived.Prop", "Base.Prop").WithLocation(8, 25)
         );
+
+        var baseComp = CreateCompilationWithRequiredMembers(@base);
+        baseComp.VerifyDiagnostics();
+
+        comp = CreateCompilation(derived, references: new[] { useMetadataReference ? baseComp.ToMetadataReference() : baseComp.EmitToImageReference() });
+        comp.VerifyDiagnostics(
+            // (4,25): error CS9501: 'Derived.Prop': cannot remove 'required' from 'Base.Prop' when overriding
```

Comment:
unrelated to this PR but it occurs to me that a diagnostic message like the following might be more consistent with other messages we have

\`'Derived.Prop' must be required because it overrides required member 'Base.Prop'\`